src/main/scala/zio/prelude/Enumerable.scala

```diff
@@ -0,0 +1,17 @@
+package zio.prelude
+
+trait Enumerable[+A] {
```

Comment:
This one could have `ordinal` and `fromOrdinal`, together with `size`, which would allow a law that ties these together to `enumerate`, as well as helpers like `min` and `max`.

Reply:
I like that.